charts/replicated-library/templates/lib/_statefulset.tpl

```diff
@@ -0,0 +1,69 @@
+{{/*
+This template serves as the blueprint for the StatefulSet objects that are created
+within the common library.
+*/}}
+{{- define "replicated-library.statefulset" }}
+  {{- $values := . -}}
+  {{- if hasKey . "AppValues" -}}
+    {{- with .AppValues.app -}}
+      {{- $values = . -}}
+    {{- end -}}
+  {{ end -}}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "replicated-library.names.appname" . }}
+  {{- with (merge ($values.labels | default dict) (include "replicated-library.labels" $ | fromYaml)) }}
+  labels: {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with (merge ($values.annotations | default dict) (include "replicated-library.annotations" $ | fromYaml)) }}
+  annotations: {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  revisionHistoryLimit: {{ $values.revisionHistoryLimit }}
+  replicas: {{ $values.replicas }}
+  podManagementPolicy: {{ default "OrderedReady" $values.podManagementPolicy }}
+  {{- $strategy := default "RollingUpdate" $values.strategy }}
+  {{- if and (ne $strategy "OnDelete") (ne $strategy "RollingUpdate") }}
+    {{- fail (printf "Not a valid strategy type for StatefulSet (%s)" $strategy) }}
+  {{- end }}
+  updateStrategy:
+    type: {{ $strategy }}
+    {{- if and (eq $strategy "RollingUpdate") $values.rollingUpdate.partition}}
+    rollingUpdate:
+      partition: {{ $values.rollingUpdate.partition }}
+    {{- end }}
+  selector:
+    matchLabels:
+      {{- include "replicated-library.labels.selectorLabels" . | nindent 6 }}
+  {{- $serviceName := default (include "replicated-library.names.appname" .) ($values.serviceName) }}
+  serviceName:  {{ $serviceName }}
```

Comment:
i think we can skip the variable here since we're only using this in one spot. just use the `default` function right on `serviceName`. `serviceName: {{ default ...`